pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -524,21 +540,35 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				set.Namespace,
 				set.Name,
 				replicas[target].Name)
-			err := ssc.podControl.DeleteStatefulPod(set, replicas[target])
+			if err := ssc.podControl.DeleteStatefulPod(set, replicas[target]); err != nil {
```

Comment:
@alculquicondor  i am not sure if thats necessary. Are you asking to call get pods again and then verify the error returned is IsNotFound  or on the error returned by DeletePod ? I think if DeletePod did not return any error, we can be confident, the pod got deleted

Reply:
I'm asking you to check the error returned by `DeleteStatefulPod`. It would return an error (of type `NotFound`) if the Pod was already deleted. You can assume such error as success in this case.

Reply:
done